src/main/ngapp/accio/src/app/top-bar/top-bar.component.ts

```diff
@@ -6,14 +6,24 @@ import { Component, OnInit } from '@angular/core';
   styleUrls: ['./top-bar.component.css']
 })
 export class TopBarComponent implements OnInit {
-
   constructor() { }
 
-  ngOnInit(): void {
-  }
+  userSignedIn = false;
+  ngOnInit(): void {}
 
-  loggedIn = false;
-  toggleButton() {
-    this.loggedIn = !this.loggedIn;
+  // this needs to be fixed. 
+
+  toggleButton(): string {
+    if (!this.userSignedIn) { 
+      console.log('User Signed in');
+      this.userSignedIn = true;
+      return signIn;
+    }
+    console.log('User Signed out');
+    this.userSignedIn = false;
+    return signOut;
```

Comment:
Out of curiosity, does this return value get used? I see this function used as a callback, but it's not clear to me what the return value does.

Reply:
The return value sets what the button says on it. Currently, the button only displays SIGN_IN because the function isn't working properly. I also updated the comment to say this, but because the button doesn't really matter until the User API is set up I pushed off fixing it for now to focus on integrating the main algorithm.